### PR TITLE
Attempt to make API errors a little more human usable

### PR DESCRIPTION
**Pull request by april**

### CODE REVIEW

File: httpobs/website/api.py [hunk 1]
```diff
@@ -24,15 +24,28 @@ def api_post_scan_hostname():
...
     # Fail if it's not a valid hostname (not in DNS, not a real hostname, etc.)
+    ip = True if valid_hostname(hostname) is None else False
     hostname = valid_hostname(hostname) or valid_hostname('www.' + hostname)  # prepend www. if necessary
-    if not hostname:
-        return {'error': '{hostname} is an invalid hostname'.format(hostname=request.args.get('host', ''))}
+
+    if ip:
+        return {
+            'error': 'invalid-hostname-ip',
+            'text': '{hostname} is an IP address, not hostname'.format(hostname=request.args.get('host', '')),
```
Comment (floatingatoll) [inline httpobs/website/api.py]: `'"{hostname}"` may read better if `get('host', '')` returns ''.

**Comment by april** [comment]
I'm not sure I understand your comment?

File: httpobs/website/api.py (hunk 1 above)
Comment (april) [inline httpobs/website/api.py]: I'm not sure I understand, but it's not the best worded. I'll try to make it more readable.

Comment (Nightfirecat) [inline httpobs/website/api.py]: He's saying that `hostname` could be an empty string if `get('host', '')` returns an empty string, so surrounding it with quotes in the return text would make that clear.

Comment (april) [inline httpobs/website/api.py]: Oh, got'cha. In any case, I've already changed this particular error so it's likely not going to be a problem and most things should be using the error code anyways.  :)

Comment (Nightfirecat) [inline httpobs/website/api.py]: It is still used in a couple other places in that same file:
* [line 38](https://github.com/mozilla/http-observatory/blob/master/httpobs/website/api.py#L38)
* [line 68](https://github.com/mozilla/http-observatory/blob/master/httpobs/website/api.py#L68)
* [line 76](https://github.com/mozilla/http-observatory/blob/master/httpobs/website/api.py#L76)
* [line 104](https://github.com/mozilla/http-observatory/blob/master/httpobs/website/api.py#L104)

I agree on your point about using error codes instead.

**Pull request merged**